### PR TITLE
General Refactor

### PR DESCRIPTION
**Pull request by ollyau**

### CODE REVIEW

File: Source/Conversions.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#pragma once
```
Comment (StevenFrost) [inline Source/Conversions.h]: Maybe use the tools from libaw? We could even think about making these sort of utilities common but let's just copy for now.

File: Source/SimCache.h [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
-    std::string GetName() { return m_name; }
-    double GetDistance(VectorR3 position) { return (m_positionCartesian - position).Norm(); }
-
-private:
-    std::string m_name;
-    VectorR3 m_positionCartesian;
-    double m_acquireDistanceMeters;
-};
\ No newline at end of file
+#pragma once
```
Comment (StevenFrost) [inline Source/SimCache.h]: Call this `Cache.h`. Let's keep SimCache.cpp only as the main entry point for the DLL.

File: Source/SimCache.cpp [hunk 3]
```diff
@@ -1,468 +1,515 @@
...
+    IGaugeCCallback* CreateGaugeCCallback()
+    {
+        return new SIMCACHEGaugeCallback(GetContainerId());
+    }
+};
+
+//
+// PanelCallback Override
+//
+class SIMCACHEPanelCallback : public PanelCallback
```
Comment (StevenFrost) [inline Source/SimCache.cpp]: Move all these classes out, refactor them & move to well named files etc. This file should only contain things that are needed for DLL setup & main.

File: Source/SimCache.vcxproj [hunk 4]
```diff
@@ -1,79 +1,84 @@
...
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <ImportGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup>
+    <_ProjectFileVersion>12.0.30501.0</_ProjectFileVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
```
Comment (StevenFrost) [inline Source/SimCache.vcxproj]: For @StevenFrost, change the build setup.

File: Source/SimCacheImpl.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
+#include "Conversions.h"
```
Comment (StevenFrost) [inline Source/SimCacheImpl.cpp]: Call this Cache.cpp

File: Source/SimCacheManager.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        static Manager instance;
+        return instance;
+    }
+
+    Manager(Manager const&) = delete;
+    void operator=(Manager const&) = delete;
+
+    void AddCache(ICache::Ptr cacheToAdd);
+
+    void IndexIncrement() { m_index = m_index + 1 > m_simCaches.size() -1 ? 0 : m_index + 1; }
```
Comment (StevenFrost) [inline Source/SimCacheManager.h]: Remove this, replace it with GetNextCache or something similar.

File: Source/Conversions.h (hunk 1 above)
Comment (ollyau) [inline Source/Conversions.h]: Not needed as discussed.  Consider dimensional analysis for additional libaw conversions.

**Comment by ollyau** [comment]
Approved!

**Comment by StevenFrost** [comment]
Approved :+1:

**Pull request merged**